src/Poseidon/Package.hs

```diff
@@ -475,43 +476,62 @@ getJointGenotypeData :: (MonadSafe m) => Bool -- ^ whether to show all warnings
 
--- | A function to create a dummy POSEIDON.yml file
--- This will take only the filenames of the provided files, so it assumes that the files will be copied into
--- the directory into which the YAML file will be written
-newPackageTemplate :: FilePath -> String -> GenotypeDataSpec -> Maybe [EigenstratIndEntry] -> Maybe [JannoRow] -> Maybe BibTeX -> IO PoseidonPackage
-newPackageTemplate baseDir name (GenotypeDataSpec format_ geno _ snp _ ind _ snpSet_) inds janno bib = do
-    (UTCTime today _) <- getCurrentTime
-    return PoseidonPackage {
+-- | A function to create a minimal POSEIDON package
+newMinimalPackageTemplate :: FilePath -> String -> GenotypeDataSpec -> PoseidonPackage
+newMinimalPackageTemplate baseDir name (GenotypeDataSpec format_ geno _ snp _ ind _ snpSet_) =
+    PoseidonPackage {
         posPacBaseDir = baseDir
-    ,   posPacPoseidonVersion = makeVersion [2, 1, 0]
+    ,   posPacPoseidonVersion = makeVersion [2, 4, 0]
     ,   posPacTitle = name
-    ,   posPacDescription = Just "Empty package template. Please add a description"
+    ,   posPacDescription = Nothing
     ,   posPacContributor = [ContributorSpec "John Doe" "john@doe.net"]
-    ,   posPacPackageVersion = Just $ makeVersion [0, 1, 0]
-    ,   posPacLastModified = Just today
+    ,   posPacPackageVersion = Nothing
+    ,   posPacLastModified = Nothing
     ,   posPacGenotypeData = GenotypeDataSpec format_ (takeFileName geno) Nothing (takeFileName snp) Nothing (takeFileName ind) Nothing snpSet_
-    ,   posPacJannoFile = Just $ name ++ ".janno"
-    -- TODO: This is not a good solution. Maybe we need pattern matching with
-    -- two different implementations of newPackageTemplate depending on whether
-    -- the input janno object is Nothing or not
-    ,   posPacJanno =
-            case janno of
-                Nothing -> case inds of
-                    Nothing -> throw $ PoseidonNewPackageConstructionException "Missing Individual- and Group IDs. This should never happen"
-                    Just a -> createMinimalJanno a
-                Just a -> a
+    ,   posPacJannoFile = Nothing
+    ,   posPacJanno = []
     ,   posPacJannoFileChkSum = Nothing
-    ,   posPacBibFile = Just $ name ++ ".bib"
-    ,   posPacBib =
-            case bib of
-                Nothing -> [] :: BibTeX
-                Just a  -> a
+    ,   posPacBibFile = Nothing
+    ,   posPacBib = [] :: BibTeX
     ,   posPacBibFileChkSum = Nothing
     ,   posPacReadmeFile = Nothing
     ,   posPacChangelogFile = Nothing
     ,   posPacDuplicate = 1
     }
 
+-- | A function to create a more complete POSEIDON package
+-- This will take only the filenames of the provided files, so it assumes that the files will be copied into
+-- the directory into which the YAML file will be written
+newPackageTemplate :: FilePath -> String -> GenotypeDataSpec -> Maybe (Either [EigenstratIndEntry] [JannoRow]) -> Maybe BibTeX -> IO PoseidonPackage
```

Comment:
Note that it should never be necessary to use `Maybe` together with List types, since a list already is an extended Maybe, with `[]` corresponding to `Nothing`. There is no semantic difference between `Nothing` and `[]`, so in fact such a type choice creates multiple options to express the same thing.

Reply:
Alright - I replaced `Maybe Bibtex` with only `Bibtex`. Or did you mean I should also refactor `Maybe (Either [EigenstratIndEntry] [JannoRow])`? I was actually a bit proud of the neat pattern matching it allows then

Reply:
Hmm. Yeah, OK, I get now what you're doing with that `Maybe Either` type. Well, for me it's OK to keep it, but I think I would have solved that slightly differently. I would have just taken `Either [EigenstratIndEntry] [JannoRow]` and then have both `Left []` and `Right []` being interpreted as `Nothing`. I find that slightly cleaner... but it's partly up to taste.

By the way, what happens if the user provides the wrong number of entries here? Is that checked or can that not happen because of upstream or downstream checks?

Reply:
Hm - I think I like my interface more. With your implementation the NULL case becomes slightly awkward, because you have to call the function with `Left []` or `Right []` when you actually want to express `Nothing`.

There are no number checks here. The function is used within forge and init and imho not vulnerable there, because the input is well prepared. But these functions are pretty useful also outside of trident, so maybe we should indeed add some more input checks :thinking:. Whenever I come across a case like that I remember Futharks [Size Types](https://futhark.readthedocs.io/en/latest/language-reference.html#size-types)!

Reply:
Yes all good then!